.github/workflows/os-osd-deployment.yml

```diff
@@ -0,0 +1,131 @@
+name: Deploy OpenSearch and OpenSearch Dashboards 
+
+on:
+  push:
+    branches: [ main ]
+  # adds workflow_dispatch for manually running a workflow
+  workflow_dispatch:
+
+jobs:
+
+  Pre-Deployment:
+    runs-on: ubuntu-latest
+
+    outputs:
+      config_change_dev: ${{steps.filter.outputs.dev}}
+      config_change_prod: ${{steps.filter.outputs.prod}}
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          # creats filters for tracking dev and prod changes
+          filters: |
+            dev:
+              - 'config/playground/helm/dev/**'
+            prod:
+              - 'config/playground/helm/prod/**'
+
+  OS-OSD-Dev-Deployment:
+    if: ${{ needs.Pre-Deployment.outputs.config_change_dev == 'true'}}
+    runs-on: ubuntu-latest
+
+    needs: Pre-Deployment
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Step 1 - Dev - Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_DEV }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_DEV }}
+          aws-region: ${{ secrets.AWS_REGION_DEV }}
+
+      - name: Step 2 - Dev - Deploy OpenSearch and OpenSearch Dashboards By Helm Chart
+        uses: elastic-analytics/dashboards-action@main
+        env:
+          KUBE_CONFIG_DATA: ${{ secrets.KUBE_CONFIG_DATA_DEV }}
+        with:
+          plugins: "" # optional, list of Helm plugins. eg. helm-secrets or helm-diff.
+          # lists couple helm and kubetl commands here as placeholder, will add more
+          # detail deployment command when helm configuration file ready.
+          command: |
+            helm repo add opensearch https://opensearch-project.github.io/helm-charts/
+            helm repo update
+            helm search repo opensearch
+            kubectl get nodes
+
+  OS-OSD-Prod-Deployment:
+    if: ${{ needs.Pre-Deployment.outputs.config_change_prod == 'true'}}
+    runs-on: ubuntu-latest
+
+    needs: Pre-Deployment
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Step 1 - Prod - Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_PROD }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_PROD }}
+          aws-region: ${{ secrets.AWS_REGION_PROD }}
+
+      - name: Step 2 - Prod - Deploy OpenSearch and OpenSearch Dashboards By Helm Chart
+        uses: elastic-analytics/dashboards-action@main
+        env:
+          KUBE_CONFIG_DATA: ${{ secrets.KUBE_CONFIG_DATA_PROD }}
+        with:
+          plugins: "" # optional, list of Helm plugins. eg. helm-secrets or helm-diff.
+          command: |
+            helm repo add opensearch https://opensearch-project.github.io/helm-charts/
```

Comment:
if we use helm to deploy os/osd, could we support deploy source code change which by choose a specific repo/branch?

Reply:
it need additional work for pushing the changed source code to container, it will be used for helm chart registration